aws/data_source_aws_workspaces_image_test.go

```diff
@@ -0,0 +1,50 @@
+package aws
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccDataSourceAwsWorkspacesImage_basic(t *testing.T) {
+	dataSourceName := "data.aws_workspaces_image.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccWorkspacesImagePreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsWorkspacesImageConfig(os.Getenv("AWS_WORKSPACES_IMAGE_ID")),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "image_id", "wsi-vwtykwqyx"),
+					resource.TestCheckResourceAttr(dataSourceName, "name", "Windows10Baseline"),
+					resource.TestCheckResourceAttr(dataSourceName, "description", "Google Chrome AWS WorkDocs Drive CrowdStrike Falcon Russian/Ukrainian Language Pack"),
+					resource.TestCheckResourceAttr(dataSourceName, "os", "WINDOWS"),
+					resource.TestCheckResourceAttr(dataSourceName, "required_tenancy", "DEFAULT"),
+					resource.TestCheckResourceAttr(dataSourceName, "state", "AVAILABLE"),
```

Comment:
Since we're retrieving a WorkSpaces Image using an arbitrary ID, we won't know these values beforehand. You might be able to test this by retrieving the API resource by ID, and then comparing the values in the Terraform resource state. `testAccCheckEfsMountTarget()` is an example of the approach you could take

Reply:
`testAccCheckEfsMountTarget()` retrieves actual resource attrubtes within AWS API, but doesn't test them against state 😄  But I got the idea, you meant item 2 at [Anatomy of an Acceptance Test](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md#anatomy-of-an-acceptance-test) – `testAccCheckCloudWatchDashboardExists()`.

Reply:
Fixed